src/__tests__/updater.js

```diff
@@ -0,0 +1,133 @@
+import { enhanceReducer } from "../updater";
+import { OFFLINE_STATUS_CHANGED, RESET_STATE } from "../types";
+import {
+  OFFLINE_COMPLETE_RETRY,
+  OFFLINE_SCHEDULE_RETRY,
+  PERSIST_REHYDRATE,
+  OFFLINE_BUSY
+} from "../constants";
+
+const initialState = {
+  busy: false,
+  lastTransaction: 0,
+  online: false,
+  outbox: [],
+  retryCount: 0,
+  retryScheduled: true,
+  netInfo: {
+    isConnectionExpensive: null,
+    reach: "NONE"
+  }
+};
+
+const config = {
+  queue: {
+    enqueue: jest.fn(),
+    dequeue: jest.fn()
+  },
+  offlineStateLens: (state: any) => {
+    debugger;
+    const { offline, ...rest } = state;
+    return {
+      get: offline,
+      set: (offlineState: any) =>
+        typeof offlineState === "undefined"
+          ? rest
+          : { offline: offlineState, ...rest }
+    };
+  }
+};
+let enReducer;
+beforeEach(() => {
+  enReducer = enhanceReducer(reducer, config);
+});
+const reducer = jest.fn(state => state);
+
+describe("buildOfflineUpdater updates for all action", () => {
+  test("test for action type: OFFLINE_COMPLETE_RETRY", () => {
+    const busyAction = () => ({
+      type: OFFLINE_BUSY,
+      payload: {
+        busy: true
+      }
+    });
+    const reducerVal = enReducer(initialState, busyAction());
+    const offlineCompleteRetry = () => ({
```

Comment:
This should be the action we're testing here, not busy 😄 (use the action creator though)